src/Sylius/Bundle/MailerBundle/Sender/Adapter/TwigSwiftMailerAdapter.php

```diff
@@ -59,19 +55,18 @@ public function __construct(
      */
     public function send(EmailInterface $email, array $recipients, array $data = array())
     {
-        $data = $this->twig->mergeGlobals($data);
-
         if (null !== $email->getTemplate()) {
+            $data = $this->twig->mergeGlobals($data);
+
             $template = $this->twig->loadTemplate($email->getTemplate());
 
             $subject = $template->renderBlock('subject', $data);
```

Comment:
Problem with this is more complicated, that API is not deprecated, and yes it's internal, but unfortunately there is no real alternative to use (interface you pointed out is [deprecated](http://twig.sensiolabs.org/doc/deprecated.html#interfaces)) so I guess it's acceptable to use it in current way till there will be official alternative.